homeassistant/components/unifi/sensor.py

```diff
@@ -398,3 +427,29 @@ def async_update_state(self, event: ItemEvent, obj_id: str) -> None:
         obj = description.object_fn(self.controller.api, self._obj_id)
         if (value := description.value_fn(self.controller, obj)) != self.native_value:
             self._attr_native_value = value
+
+        if description.is_connected_fn is not None:
+            # Send heartbeat if client is connected
+            if description.is_connected_fn(self.controller, self._obj_id):
+                self.controller.async_heartbeat(
+                    self._attr_unique_id,
+                    dt_util.utcnow() + self.controller.option_detection_time,
+                )
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await super().async_added_to_hass()
+        # Register callback for missed heartbeat
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{self.controller.signal_heartbeat_missed}_{self.unique_id}",
+                self._make_disconnected,
+            )
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect object when removed."""
+        await super().async_will_remove_from_hass()
+        # Remove heartbeat registration
+        self.controller.async_heartbeat(self._attr_unique_id)
```

Comment:
Wrap this in an if statement as the async update state `if description.is_connected_fn is not None:`